### PR TITLE
Switching to map.jinja to define platform specific values

### PR DESCRIPTION
**Pull request by daks**
Hello,

this pull request refactors actual code, to use map.jinja file to handle platform specific values. It doesn't change actual behaviour.
Could you look at it and merge it if it's ok?

Thanks

### CODE REVIEW

File: haproxy/config.sls [hunk 1]
```diff
@@ -1,13 +1,16 @@
+{% from "haproxy/map.jinja" import haproxy with context %}
+
+{% set config_file = salt['pillar.get']('haproxy:config_file_path', haproxy.config_file) %}
 haproxy.config:
  file.managed:
-   - name: {{ salt['pillar.get']('haproxy:config_file_path', '/etc/haproxy/haproxy.cfg') }}
-   - source: salt://haproxy/templates/haproxy.jinja
+   - name: {{ config_file }}
+   - source: {{ haproxy.config_file_source }}
```
Comment (n-rodriguez) [inline haproxy/config.sls]: This changes the pillar key from `haproxy:config_file_path` to `haproxy:config_file_source`

Comment (daks) [reply haproxy/config.sls]: No, it doesn't. In map.jinja config_file is the destination file on the server and config_file_source is the source on Salt master.
That means the 'ideal' pillar to override config file is 'haproxy: lookup: config_file', but to kept actual behavious there is this line https://github.com/saltstack-formulas/haproxy-formula/pull/69/files#diff-44c85aa6e52c67554f24e08baf667cfbR3 which get pillar from 'haproxy: config_file_path'

**Comment by daks** [comment]
Hello, any news about this PR?

**Comment by daks** [comment]
Hi, any updates about this PR? Can it be merged? cc @n-rodriguez @aboe76 @johnkeates

**Comment by n-rodriguez** [comment]
Sounds good to me :) @aboe76 @johnkeates ?

File: haproxy/service.sls [hunk 2]
```diff
@@ -1,7 +1,9 @@
+{% from "haproxy/map.jinja" import haproxy with context %}
+
 haproxy.service:
 {% if salt['pillar.get']('haproxy:enable', True) %}
   service.running:
-    - name: haproxy
+    - name: {{ haproxy.service }}
```
Comment (noelmcloughlin) [inline haproxy/service.sls]: Hey @daks,  This all looks good,  Just one question ...  If os_family is FreeBSD will **`haproxy.service`** be None?   There is unfortunately no defaults.yaml so formula requires mandatory pillars on FreeBSD?  Same question regarding package, etc

Comment (noelmcloughlin) [reply haproxy/service.sls]: In map.jinja only parameter defined is `group` ..

Comment (daks) [reply haproxy/service.sls]: In map.jinja, the merge uses `base='default'` to merge with default dict which itself contains `service: haproxy` so for FreeBSD (or others) `service` will be set to `haproxy`

**Comment by johnkeates** [comment]
Annoying how life sometimes gets in the way.. sorry for the delay!

**Comment by daks** [comment]
Thanks to all.

**Pull request merged**